walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,126 @@
+    // only if there's no event that starts the day.
+    if (StartOfDayIsFree(eventList) && requestHasEventAttendees) {
+      availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY,
+        firstEvent.getWhen().start(), false));
+    }
+
+    for (int i = 0; i < eventList.size() - 1; i++) {
+      TimeRange currEventTimeRange = eventList.get(i).getWhen();
+      TimeRange nextEventTimeRange = eventList.get(i + 1).getWhen();
+
+      if (!requestHasEventAttendees(request, eventList.get(i))) {
+        eventsSkipped ++;
+        continue;
+      }
+
+      if (currEventTimeRange.contains(nextEventTimeRange)) {
+        // If there is a 3rd event, connect events 1 and 3. Otherwise, connect with the end of day.
+        if (i + 2 < eventList.size() - 1) {
+          TimeRange nextnextEventTimeRange = eventList.get(i + 2).getWhen();
+          availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+            nextnextEventTimeRange.start(), true));
+          i++;
+        } else {
+          availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+            TimeRange.END_OF_DAY, true));
+        }
+      } else {
+          // If current event completely overlaps the next event, there's no reason to check
+          // the next one too, as it's start and end are completely contained.
+          if (currEventTimeRange.overlaps(nextEventTimeRange)) {
+            continue;
+          }
+
+          if (nextEventTimeRange.start() - currEventTimeRange.end() >= request.getDuration()) {
+            availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+              nextEventTimeRange.start(), false));  
+          } 
+        }
+    }
+
+    if (requestHasEventAttendees(request, lastEvent)) {
+      if (availableTimes.size() != 0){
+        if (nothingEndsTheDay(availableTimes, eventList)) {
+          availableTimes.add(TimeRange.fromStartEnd(eventList.get(eventList.size() - 1).getWhen().end(),
+            TimeRange.END_OF_DAY, true));
+        } 
+      } else {
+          if (fitsOnlyAtTheEnd(request, availableTimes, eventList)) {
+              availableTimes.add(TimeRange.fromStartEnd(eventList.get(eventList.size() - 1).getWhen().end(),
+                TimeRange.END_OF_DAY, true));
+            }
+        } 
+    } 
+
+    if (eventsSkipped == eventList.size()) {
+      availableTimes.add(TimeRange.WHOLE_DAY);
+    }
+
+    return availableTimes;
+  }
+
+  static private boolean requestHasEventAttendees(MeetingRequest request, Event currEvent) {
+    List<String> requestAttendees = new ArrayList<>(request.getAttendees());
+    List<String> eventAttendees = new ArrayList<>(currEvent.getAttendees());
+
+    return requestAttendees.stream().anyMatch(eventAttendees::contains);
+  }
+
+  // Check if the first event in our list already includes the start of the day.
+  static private boolean StartOfDayIsFree(List<Event> eventList) {
```

Comment:
method names should start with lowercase even if they're static. Also, `private static` instead of `static private`. Same for other methods